runtime/compiler/p/codegen/ForceRecompilationSnippet.cpp

```diff
@@ -47,6 +47,9 @@ uint8_t *TR::PPCForceRecompilationSnippet::emitSnippetBody()
    uint8_t             *buffer = cg()->getBinaryBufferCursor();
    TR::SymbolReference  *induceRecompilationSymRef = cg()->symRefTab()->findOrCreateRuntimeHelper(TR_PPCinduceRecompilation);
    intptr_t startPC = (intptr_t)((uint8_t*)cg()->getCodeStart());
+   // The relo runtime expects this kind of address to be zero in a relocatable binary, since it uses
+   // |= to update the address chunks. See TR_PPC64RelocationTarget::storeAddressSequence.
+   intptr_t startPCRegValue = cg()->needRelocationsForCurrentMethodStartPC() ? 0 : startPC;
```

Comment:
I'm not sure if this causes a problem, but there is actually a query `cg()->canEmitDataForExternallyRelocatableInstructions()` in power codegen that determines whether or not to emit these addresses directly. That query is used in the higher-level `loadAddressConstant` family of functions that we'd ideally be able to use here (but can't). That should probably be used here.

Reply:
That's what I was thinking